### PR TITLE
[Test] Add flatbuffer converter and decoder test

### PR DESCRIPTION
**Pull request by gichan-jang**
Add flatbuffer converter and decoder run test

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: tests/nnstreamer_flatbuf/runTest.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+do_test BGRx 640 480 2-1
+do_test RGB 640 480 2-2
+do_test GRAY8 640 480 2-3
+do_test BGRx 642 480 2-4
+do_test RGB 642 480 2-5
+do_test GRAY8 642 480 2-6
+
+# Fail Test: YUV is given
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=YUV,width=280,height=40,framerate=0/1 ! videoconvert ! video/x-raw, format=YUV ! tensor_converter silent=TRUE ! tensor_decoder mode=flatbuf ! other/flatbuf-tensor ! tensor_converter ! filesink location=\"test.yuv.fail.log\" sync=true" 5F_n 0 1 $PERFORMANCE
```
Comment (dongju-chae) [inline tests/nnstreamer_flatbuf/runTest.sh:66]: Minor comment: here, the testcase number starts from `5F_n`, but the previous testcase number was `2-6`

Comment (gichan-jang) [reply tests/nnstreamer_flatbuf/runTest.sh]: Right. Updated!

File: tests/nnstreamer_flatbuf/runTest.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    echo "Test Case Generation Started"
+    python ../nnstreamer_converter/generateGoldenTestResult.py
+    python ../nnstreamer_merge/generateTest.py
+    sopath=$1
+fi
+convertBMP2PNG
+
+PATH_TO_PLUGIN="../../build"
+
+gst-launch-1.0 -m -v --gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=RGB,width=280,height=40,framerate=0/1 ! tee name=t ! queue ! videoconvert ! video/x-raw, format=BGRx ! tensor_converter silent=TRUE ! filesink location="test.bgrx.log" sync=true t. ! queue ! filesink location="test.rgb.log" sync=true
```
Comment (myungjoo) [inline tests/nnstreamer_flatbuf/runTest.sh:36]: why do you need this? (LINE 36) Its output is overwritten by LINE 38.

File: tests/nnstreamer_flatbuf/runTest.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=RGB,width=280,height=40,framerate=0/1 ! tee name=t ! queue ! videoconvert ! video/x-raw, format=BGRx ! tensor_converter silent=TRUE ! tensor_decoder mode=flatbuf ! other/flatbuf-tensor ! tensor_converter ! filesink location=\"test.bgrx.log\" sync=true t. ! queue ! filesink location=\"test.rgb.log\" sync=true" 1R 0 0 $PERFORMANCE
+
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=GRAY8,width=280,height=40,framerate=0/1 ! queue ! tensor_converter silent=TRUE ! tensor_decoder mode=flatbuf ! other/flatbuf-tensor ! tensor_converter ! filesink location=\"test.gray8.log\" sync=true" 1G 0 0 $PERFORMANCE
+
+callCompareTest testcase01.bgrx.golden test.bgrx.log 1-1 "BGRX Golden Test" 1 0
+callCompareTest testcase01.rgb.golden test.rgb.log 1-2 "RGB Golden Test" 1 0
+callCompareTest testcase01.gray8.golden test.gray8.log 1-3 "Gray8 Golden Test" 1 0
```
Comment (myungjoo) [inline tests/nnstreamer_flatbuf/runTest.sh:44]: videotestsrc includes randomely generated pixels. How do you create proper testcase01.*.golden?

Comment (gichan-jang) [reply tests/nnstreamer_flatbuf/runTest.sh:44]: You're right.. It's weird how it passed. Maybe the first frame of videotestsrc is consistent?
I will change videotestsrc to filesrc and related other tests (tensor converter etc.)

File: tests/nnstreamer_flatbuf/runTest.sh (hunk 2 above)
Comment (gichan-jang) [reply tests/nnstreamer_flatbuf/runTest.sh:36]: My bad.. I will remove it. Thank you.

File: tests/nnstreamer_flatbuf/runTest.sh (hunk 3 above)
Comment (dongju-chae) [reply tests/nnstreamer_flatbuf/runTest.sh:44]: IMO, `videotestsrc` generates deterministic random values, and there're some testcases using `videotestsrc` and its golden data.

Comment (anyj0527) [reply tests/nnstreamer_flatbuf/runTest.sh:44]: > IMO, `videotestsrc` generates deterministic random values, and there're some testcases using `videotestsrc` and its golden data.

Right, this is the case: https://github.com/nnstreamer/nnstreamer/blob/49fa3a91ef5749ae834c079a2d66e4fa78a46510/tests/nnstreamer_decoder_image_segment/runTest.sh#L77-L79

Comment (kparichay) [reply tests/nnstreamer_flatbuf/runTest.sh:42]: > IMO, `videotestsrc` generates deterministic random values, and there're some testcases using `videotestsrc` and its golden data.

Yes, `videotestsrc` starts with a fixed seed of 0 and update as the function below (all deterministic it seems)
https://github.com/GStreamer/gst-plugins-base/blob/648acbe726a74b547d6a4003265e492cefe69a9f/gst/videotestsrc/videotestsrc.c#L38-L42

Comment (myungjoo) [reply tests/nnstreamer_flatbuf/runTest.sh:42]: Let's not assume that videotestsrc's randomly generated pixels will always be exactly repeated with every restart.

Refer to https://gstreamer.freedesktop.org/documentation/videotestsrc/index.html?gi-language=c , use "pattern" parameter to set some "complex pattern" without random pixels.

File: tests/nnstreamer_flatbuf/runTest.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+## @brief SSAT Test Cases for flatbuffers subplugin of tensor converter and decoder
+## @details After decoding the tensor into flatbuffer, convert it to tensor(s) again to check if it matches the original
+##
+if [[ "$SSATAPILOADED" != "1" ]]; then
+    SILENT=0
+    INDEPENDENT=1
+    search="ssat-api.sh"
+    source $search
+    printf "${Blue}Independent Mode${NC}
+"
```
Comment (wooksong) [inline tests/nnstreamer_flatbuf/runTest.sh:17]:
```suggestion
    printf "${Blue}Independent Mode${NC}"
```

Comment (gichan-jang) [reply tests/nnstreamer_flatbuf/runTest.sh]: Thank you. Updated!

File: tests/nnstreamer_flatbuf/runTest.sh (hunk 3 above)
Comment (gichan-jang) [reply tests/nnstreamer_flatbuf/runTest.sh]: I removed the randomly generated patterns and changed to pattern no.13.

Screenshot from 2020-07-01 13-38-25

**Pull request merged**